src/main/java/com/google/sps/servlets/ProfileCreationServlet.java

```diff
@@ -55,7 +55,8 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     throws ServletException, IOException {
 
     HttpSession session = request.getSession();
-    if (((int) session.getAttribute("isLoggedIn"))== 1) {
+    Integer isLoggedIn = (Integer) session.getAttribute("isLoggedIn");
+    if ((isLoggedIn != null)&&(isLoggedIn == 1)) {
```

Comment:
If "isLoggedIn" is only set when the user is logged in, then you can skip the "isLoggedIn == 1)" check.

Also, run the formatter, this should be: 

if ((isLoggedIn != null) && (isLoggedIn == 1)) {